presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -793,6 +801,18 @@ public ListeningExecutorService createResourceManagerExecutor(ResourceManagerCon
         binder.bind(NodeStatusNotificationManager.class).in(Scopes.SINGLETON);
 
         binder.bind(PlanCheckerProviderManager.class).in(Scopes.SINGLETON);
+
+        // Worker session property providers
+        MapBinder<String, WorkerSessionPropertyProvider> mapBinder =
+                newMapBinder(binder, String.class, WorkerSessionPropertyProvider.class);
+        mapBinder.addBinding("java-worker").to(JavaWorkerSessionPropertyProvider.class).in(Scopes.SINGLETON);
```

Comment:
We need these even with sidecar enabled, because in our environment queries can bounce between java and native clusters, so users may specify properties for both.  We want to hide those properties, but we don't want the queries to fail.

Reply:
How about we make the default false (disable the properties), but add a config flag that turns it on if needed?

Reply:
Yes, that is the plan. 
Opened an issue for the same: https://github.com/prestodb/presto/issues/23827
Can I add this support in a follow-up PR?

Reply:
SGTM!